js/plugins/vertexai/src/gemini.ts

```diff
@@ -619,7 +721,7 @@ export function defineGeminiModel(
 
       // Handle system instructions separately
       let systemInstruction: Content | undefined = undefined;
-      if (SUPPORTED_V15_MODELS[name]) {
+      if (modelInfo?.supports?.systemRole) {
```

Comment:
See comment above. I think we had an intentional split where system roles were supported via simulation in 1.0 era models, and via systemInstructions in 1.5+ era models.